arraycontext/impl/pyopencl.py

```diff
@@ -43,6 +43,12 @@
 from arraycontext.container import serialize_container, is_array_container
 from arraycontext.context import ArrayContext
 
+try:
+    import pyopencl as cl  # noqa: F401
+    import pyopencl.array as cl_array
+except ImportError:
+    pass
```

Comment:
Thanks for thinking about this. I don't think this suffices, unfortunately. Importing pyopencl is relatively heavyweight, and we should wait to do it until the user asks us to do so. As a result, I think the best option is to move the numpy bits to a separate file.

Reply:
But wouldn't we end up doing:
```
from arraycontext.impl.pyopencl.fake_numpy import maximum, ...
```
everywhere? i.e. I think we will still keep on adding a line before every math routine invocation.

Reply:
Not sure I understand. In my view, the only place that imports from that submodule would be the `.np` property.

Reply:
Aah right. I understand and I think this will work. Thanks!